### ISSUE TITLE
Faster builds: Decoupled `mondo` repo artefacts

### ISSUE THREAD

**Issue by joeflack4** [monarch-initiative/mondo-ingest#619]

**Comment by matentzn** [comment]
Please assign this to me, this should not be happening

**Comment by joeflack4** [comment]
@matentzn I marked it as a low priority; it's just an idea I came up with and the details kinda poured out of my head. We can discuss some other time. I am curious to what you mean by "this should not be happening". There is no bug here. This is an optimization.

**Comment by matentzn** [comment]
I thought we just spend a lot of resources fixing this: https://github.com/monarch-initiative/mondo-ingest/pull/594/files

**Comment by joeflack4** [comment]
Ah, this idea improves upon that further. 

What we have done already is making so that whenever you run commands that use mondo.owl etc, it will always check and rebuild them if they are out of date. 

What this issue proposes is a decoupled process that uses a GitHub action to build these regularly and then upload them to get up releases. Then, when you run a goal which needs mondo.owl, it will download them if the local files are out of date.

**Comment by matentzn** [comment]
I guess if the process consumes that much time this is a possibility, but given the fact that you may need to rerun from the repo from time to time anyways (for example if you need a change that was just pushed) it introduces a bit of complexity in the code. The PR I linked (please ensure it is seen thru as I dont have eyes on it) should make sure that the code is only run once for any state of the Mondo `master` branch. But you suggestion also makes some sense; basically you have actions run every night rebuilding mondo.owl and mondo.sssom.tsv, caches them and the ingest code just pulls them (not sure how to get reliable download URLs from GH actions, I remember this was problematic in the past)

**Comment by joeflack4** [comment]
>  introduces a bit of complexity

True. If you are running a goal that needs `mondo.owl` etc, and it detects a change, it'll try to `wget` from the release, and if the build process is still running, then without any additional logic to account for this it will just download the old release.

>  The PR I linked (please ensure it is seen thru

Oh, [Refactor git refresh system (small refactor) #594](https://github.com/monarch-initiative/mondo-ingest/pull/594) didn't have an assignee. It looks like you got stuck on an issue during your build. I've assigned this to myself now, but it is a low priority.

> should make sure that the code is only run once for any state of the Mondo master branch

No worries; that has already been completed in:
- #551

> not sure how to get reliable download URLs from GH actions

I think you mean releases? I haven't had any such problems. We have ICD11 and OMIM pipelines set up using GitHub releases, and there's been no issue yet.

**Comment by matentzn** [comment]
> I think you mean releases? I haven't had any such problems. We have ICD11 and OMIM pipelines set up using GitHub releases, and there's been no issue yet.

Yes but you have specific repos for that. Are you suggesting to create a mondo.owl nightly built _repo_ so you can create github releases? As if it was a "source"? I thought you meant to create a github action in the mondo-ingest repo that caches the built files in the GHA cache. In any case, maybe your idea of a separate repo isnt a bad one; Just make sure its raised it a call with Trish to discuss pros and cons.

> Oh, https://github.com/monarch-initiative/mondo-ingest/pull/594 didn't have an assignee.

Dropped the ball on this and I wont be able to pick it up again. Maybe just try to review the diff carefully until you understand the point of all the changes, and if you dont understand something / or want to question a modelling decision, just let me know?

**Comment by joeflack4** [comment]
I'll talk about #594 w/ Trish and prioritize.

For `mondo.owl` etc as releases, I proposed 3 alternatives in the OP:
> ### Possible implementation details
> #### Where to store releases
> - a. Release these somehow in the `mondo` repo releases
> - b. Separate repo just for these artefacts
> - c. Add to `mondo-ingest` releases.

I'd rather not do (b) and create a separate repo, but I think it is the easiest solution.

It would run more often than nightly. In TermHub, we run things every 20 minutes. It will check if there are any updates to `main`, and if not, it will do nothing.

**Comment by matentzn** [comment]
I am neither opposed nor strongly in favour, I think the idea is generally worth considering as it could shave off 30 min of every build we do..
